activation/poet.go

```diff
@@ -297,7 +314,13 @@ func (c *HTTPPoetClient) req(ctx context.Context, method, path string, reqBody,
 	}
 	req.Header.Set("Content-Type", "application/json")
 
-	res, err := c.client.Do(req)
+	var res *http.Response
+
+	if path == submitPath {
+		res, err = c.submitChallengeClient.Do(req)
+	} else {
+		res, err = c.client.Do(req)
+	}
```

Comment:
Instead of selecting on the path, I think it would be better to pass the client that should be used to `req`. The caller can decide which client is used depending on multiple circumstances, instead of relying on `req` and the path.